### PR TITLE
NMS-12203: Upgrade Kafka components to 2.3.0

### PR DESCRIPTION
**Pull request by Naicisum**
NMS-12203: Upgrade Kafka components to 2.3.0

### All Contributors

* [x] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [x] Have you made a comment in that issue which points back to this PR?
* [x] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

Once the reviewer(s) accept the PR and the branch passes continuous integration in Bamboo, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12203
* Bamboo (Continuous Integration): https://bamboo.opennms.org/

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1290,13 +1290,13 @@
...
     <resilience4jVersion>0.17.0</resilience4jVersion>
-    <rocksdbjniVersion>5.15.10</rocksdbjniVersion>
+    <rocksdbjniVersion>5.18.3</rocksdbjniVersion>
     <scalaLibraryVersion>2.12.8</scalaLibraryVersion>
     <scalaLoggingVersion>3.9.0</scalaLoggingVersion>
     <scalaVersion>2.12</scalaVersion>
     <servicemixSpecsVersion>2.9.0</servicemixSpecsVersion>
     <servletApiVersion>3.1.0</servletApiVersion>
-    <slf4jVersion>1.7.21</slf4jVersion>
+    <slf4jVersion>1.7.26</slf4jVersion>
```
Comment (cgorantla) [inline pom.xml]: Don't see this is getting used in kafka related features. Any other reason this is upgraded ?

Comment (Naicisum) [reply pom.xml]: > Smoke tests seems to be failing. Can you take a look once ?

Smoke failed in CircleCI due to an issue with CircleCI, and/or a known issue with Karaf loading dependencies. The smoke passed on Bamboo, didn't want to waste the minutes triggering it again on the off chance it fails due to the CircleCI/Karaf known issues.

Comment (Naicisum) [reply pom.xml]: > Don't see this is getting used in kafka related features. Any other reason this is upgraded ?

See: https://search.maven.org/artifact/org.apache.servicemix.bundles/org.apache.servicemix.bundles.kafka_2.12/2.3.0_1/bundle

Pom lists that version in the bundle, this was my reason for upgrading that particular version.

Comment (cgorantla) [reply pom.xml]: Since all smoke tests are passing, I would think it is safe.

**Comment by cgorantla** [comment]
@Naicisum I think it would be good to separate CircleCI improvements into different PR.

**Comment by Naicisum** [comment]
Back to the basics, these were the original changes.

**Comment by Naicisum** [comment]
All tests passed, if you click on details for `smoke-test-full` you will see the job marked at `Circle Bug` but the container status of all 8 are green.

**Pull request merged**